core/src/main/java/org/fao/geonet/kernel/search/LuceneQueryBuilder.java

```diff
@@ -960,21 +960,16 @@ private void addBBoxQuery(Map<String, Set<String>> searchCriteria, BooleanQuery
     private void addPrivilegeQuery(LuceneQueryInput luceneQueryInput, BooleanQuery query) {
         // Set user groups privileges
         Set<String> groups = luceneQueryInput.getGroups();
-        String editable$ = luceneQueryInput.getEditable();
-        boolean editable = BooleanUtils.toBoolean(editable$);
+        Set<String> editableGroups = luceneQueryInput.getEditableGroups();
         BooleanQuery groupsQuery = new BooleanQuery();
         boolean groupsQueryEmpty = true;
         BooleanClause.Occur groupOccur = LuceneUtils.convertRequiredAndProhibitedToOccur(false, false);
-        if (!CollectionUtils.isEmpty(groups)) {
-            for (String group : groups) {
+
+        if (!CollectionUtils.isEmpty(editableGroups)) {
+        	Log.trace(Geonet.SEARCH_ENGINE, "We have editable groups to add");
+            for (String group : editableGroups) {
                 if (StringUtils.isNotBlank(group)) {
-                    if (!editable) {
```

Comment:
Instead of taking groups from user and add view privileges or editing privileges chaotically based on if the user is editor of any of their groups, now we have two separated lists. One specifically for groups with editing privileges.